tests/scripts/all.sh

```diff
@@ -920,38 +920,32 @@ can_run_cc_output() {
     $result
 }
 
-can_run_arm_linux_gnueabi=
+can_run_arm_linux_gnueabi=can_run_arm_linux_gnueabi
```

Comment:
It was a cache before 7bc7bd1010557ffe8ed6fc078e67be19d2c07093 but not now.

Reply:
It's still a cache, the intention is to just use the variable directly from `support_` functions instead of calling the function directly.

```shell
root@76ab798e8d6a:/var/test# bash -evx test.sh

: ${ARM_LINUX_GNUEABIHF_GCC_PREFIX:=arm-linux-gnueabihf-}
+ : arm-linux-gnueabihf-

can_run_cc_output() {
    cc="$1"
    result=false
    if type "$cc" >/dev/null 2>&1; then
        testbin=$(mktemp)
        if echo 'int main(void){return 0;}' | "$cc" -o "$testbin" -x c -; then
            if "$testbin" 2>/dev/null; then
                result=true
            fi
        fi
        rm -f "$testbin"
    fi
    $result
}

can_run_arm_linux_gnueabihf=can_run_arm_linux_gnueabihf
+ can_run_arm_linux_gnueabihf=can_run_arm_linux_gnueabihf
can_run_arm_linux_gnueabihf () {
    if can_run_cc_output "${ARM_LINUX_GNUEABIHF_GCC_PREFIX}gcc"; then
        can_run_arm_linux_gnueabihf=true
    else
        can_run_arm_linux_gnueabihf=false
    fi
    $can_run_arm_linux_gnueabihf
}

support_test_arm_linux_gnueabihf_gcc_armv7 () {
    $can_run_arm_linux_gnueabihf
}

support_test_arm_linux_gnueabihf_gcc_thumb_2 () {
    $can_run_arm_linux_gnueabihf
}

support_test_arm_linux_gnueabihf_gcc_armv7
+ support_test_arm_linux_gnueabihf_gcc_armv7
+ can_run_arm_linux_gnueabihf
+ can_run_cc_output arm-linux-gnueabihf-gcc
+ cc=arm-linux-gnueabihf-gcc
+ result=false
+ type arm-linux-gnueabihf-gcc
++ mktemp
+ testbin=/tmp/tmp.897dZr57b7
+ echo 'int main(void){return 0;}'
+ arm-linux-gnueabihf-gcc -o /tmp/tmp.897dZr57b7 -x c -
+ /tmp/tmp.897dZr57b7
+ result=true
+ rm -f /tmp/tmp.897dZr57b7
+ true
+ can_run_arm_linux_gnueabihf=true
+ true

support_test_arm_linux_gnueabihf_gcc_thumb_2
+ support_test_arm_linux_gnueabihf_gcc_thumb_2
+ true
```

Reply:
Okay, right, I see what's happening. So it means instead of

```
if [ $can_run_arm_linux_gnueabihf ]; then
    ...
fi
```

it's important to put

```
if $can_run_arm_linux_gnueabihf; then
    ...
fi
```

This is actually quite different to most of the code in `all.sh` - when testing shell variables we tend to use `[` (e.g. tests of `$QUIET`, `$base_config`, `$item`, etc) but when using functions we do that without the `[` (e.g. for `in_mbedtls_repo`, `support_$component`, etc).

Since doing the test the wrong way will lead to incorrect results, and since we have both variables and functions with the same name, we either need one very good explanatory comment, or to do this a different way.

My preference would be to always call a function, and have it cache the value itself:

```
can_run_arm_linux_gnueabihf()
{
    if [ -z ${can_run_arm_linux_gnueabihf+x} ]; then
        # set can_run_arm_linux_gnueabihf
    fi
    $can_run_arm_linux_gnueabihf
}
```

Reply:
Ok, in that case I think I'll force-push this PR to back out the last commit.

Reply:
Thanks - note I think we just want the function, and the shell variable of the same name can be internal to that function, so callers don't have to bother with it (simplicity)